### DISCUSSION TITLE
Issue with next-intl locale switching for one colleague

### DISCUSSION THREAD

**Discussion by swajp** [amannn/next-intl#1554] [General]
Hi there,

I’m currently using next-intl in my Next.js project to handle language switching, and I’m encountering a peculiar issue that only affects one of my colleagues.
	Configuration:
	•	defaultLocale is set to en
	•	localePrefix is set to as-needed

The problem:
When my colleague attempts to switch the language from en to cs, the /cs prefix gets removed from the URL, redirecting them back to /, and the language switches back to English (en). This issue doesn’t occur for me or any other team members.

For example:
	•	Expected behavior: Switching from /en to /cs should update the URL to include /cs and load the page in Czech.
	•	What happens for my colleague: The URL reverts to / (English default) instead of showing /cs.

We’ve checked basic things like clearing the browser cache and testing in incognito mode, but the issue persists specifically for their environment.

Also it works for them on my local server but on production or staging it doesn't.

**Comment by swajp** [comment]
I discovered that the issue is when the primary language is set to Czech and then user is not able to switch it, any idia?

**Reply by swajp** [reply]
```
export const routing = defineRouting({
    // A list of all locales that are supported
    locales: ["cs", "en"],

    // Used when no locale matches
    defaultLocale: "en",

    

    localePrefix: "as-needed"
})
```

**Reply by amannn** [reply]
Hmm, that's odd—can't see an issue with that. If `/cs` is present as a [prefix](https://next-intl-docs.vercel.app/docs/routing/middleware#location-detection-prefix), then this should be used.

Can you share a [reproduction](https://github.com/amannn/next-intl-bug-repro-app-router) with a `curl` snippet that reproduces the behavior? (should contain the `accept-language` header as it should be observable in your colleagues network tools)

**Reply by swajp** [reply]
Okay it is not just a colleague problem, it is a global. When I set my browser's primary language to Czech the url is automatically example.com/cs and when I try to remove it, it is impossible it always revert it back to English (only on production)

https://github.com/swajp/next-intl-bug-repro-app-router

(test for production)
https://next-intl-bug-repro-app-router-rouge.vercel.app

**Reply by swajp** [reply]
when the browsers language is set to english there is no problem

**Reply by amannn** [reply]
Can you please share a `curl` snippet that helps me to reproduce this? 

You can use "copy as curl" in the Chrome network tools:

![Screenshot 2024-11-20 at 17 01 27](https://github.com/user-attachments/assets/afe1a120-4e93-49af-b570-10bf3a0f7b78)

I was unfortunately not able to reproduce:

![Artboard](https://github.com/user-attachments/assets/2ac9753c-7091-4d5b-a0c3-7ac6b692337a)

**Reply by swajp** [reply]
for the repro this is the curl for the "cs"

```
curl 'https://next-intl-bug-repro-app-router-rouge.vercel.app/' \
  -H 'accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7' \
  -H 'accept-language: cs' \
  -H 'cache-control: max-age=0' \
  -H 'cookie: NEXT_LOCALE=en' \
  -H 'priority: u=0, i' \
  -H 'sec-ch-ua: "Google Chrome";v="131", "Chromium";v="131", "Not_A Brand";v="24"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "macOS"' \
  -H 'sec-fetch-dest: document' \
  -H 'sec-fetch-mode: navigate' \
  -H 'sec-fetch-site: same-origin' \
  -H 'sec-fetch-user: ?1' \
  -H 'upgrade-insecure-requests: 1' \
  -H 'user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/131.0.0.0 Safari/537.36'
```

but it seems to be working on this repro repo. but thsi is from the production web and you can see that  NEXT_LOCALE is set to en and the language  is still czech and my browser is set to czech language and it doesnt let me change the lang

```
curl 'https://www.yedem.io/cs/' \
  -H 'accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7' \
  -H 'accept-language: cs' \
  -H 'cache-control: max-age=0' \
  -H 'cookie: NEXT_LOCALE=en' \
  -H 'priority: u=0, i' \
  -H 'sec-ch-ua: "Google Chrome";v="131", "Chromium";v="131", "Not_A Brand";v="24"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "macOS"' \
  -H 'sec-fetch-dest: document' \
  -H 'sec-fetch-mode: navigate' \
  -H 'sec-fetch-site: same-origin' \
  -H 'sec-fetch-user: ?1' \
  -H 'upgrade-insecure-requests: 1' \
  -H 'user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/131.0.0.0 Safari/537.36'
```

and when i did yedem.io/en the curl is and it says status 307 and 308

```
curl 'https://www.yedem.io/en' \
  -H 'accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7' \
  -H 'accept-language: cs' \
  -H 'cookie: NEXT_LOCALE=cs' \
  -H 'priority: u=0, i' \
  -H 'sec-ch-ua: "Google Chrome";v="131", "Chromium";v="131", "Not_A Brand";v="24"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "macOS"' \
  -H 'sec-fetch-dest: document' \
  -H 'sec-fetch-mode: navigate' \
  -H 'sec-fetch-site: none' \
  -H 'sec-fetch-user: ?1' \
  -H 'upgrade-insecure-requests: 1' \
  -H 'user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/131.0.0.0 Safari/537.36'
```

```
curl 'https://www.yedem.io/en/' \
  -H 'accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7' \
  -H 'accept-language: cs' \
  -H 'cookie: NEXT_LOCALE=en' \
  -H 'priority: u=0, i' \
  -H 'sec-ch-ua: "Google Chrome";v="131", "Chromium";v="131", "Not_A Brand";v="24"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "macOS"' \
  -H 'sec-fetch-dest: document' \
  -H 'sec-fetch-mode: navigate' \
  -H 'sec-fetch-site: none' \
  -H 'sec-fetch-user: ?1' \
  -H 'upgrade-insecure-requests: 1' \
  -H 'user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/131.0.0.0 Safari/537.36'
```

**Reply by amannn** [reply]
Hmm, hard to say what's going wrong here without being able to look at the source and reproduce the error locally. Any chance you can adapt the reproduction to trigger the error case?
